otherlibs/unix/channels_win32.c

```diff
@@ -129,12 +130,10 @@ CAMLprim value caml_unix_outchannel_of_filedescr(value handle)
     caml_win32_maperr(err);
     caml_uerror("out_channel_of_descr", Nothing);
   }
-  chan = caml_open_descriptor_out(caml_win32_CRT_fd_of_filedescr(handle));
-  chan->flags |= CHANNEL_FLAG_MANAGED_BY_GC;
-                 /* as in caml_ml_open_descriptor_out() */
+  fd = caml_win32_CRT_fd_of_filedescr(handle);
   if (Descr_kind_val(handle) == KIND_SOCKET)
-    chan->flags |= CHANNEL_FLAG_FROM_SOCKET;
-  vchan = caml_alloc_channel(chan);
+    flags |= CHANNEL_FLAG_FROM_SOCKET;
+  vchan = caml_ml_open_descriptor_out_with_flags(flags,fd);
   CAMLreturn(vchan);
```

Comment:
There may be a simpler solution?  Like this:
```
  vchan = caml_ml_open_descriptor_out(fd);
  if (...) { Channel(vchan)->flags |= CHANNEL_FLAG_FROM_SOCKET;
```
(Just a suggestion.)

Reply:
The initial call to `caml_ml_open_descriptor_out` publishes the channel in the list of channels to be flushed, thus initializing the flag later can expose us to concurrent read, whereas with the new function we are not linked before the channel flags are fully initialized (and we avoid exposing the channel list to the world outside `runtime/io.c`).

Reply:
OK, that's a convincing argument, thanks.